hoot-core/src/main/cpp/hoot/core/io/HootNetworkRequest.cpp

```diff
@@ -171,42 +206,13 @@ bool HootNetworkRequest::_networkRequest(const QUrl& url, int timeout,
     //  Replace the IP address in the error string with <host-ip>
     HootNetworkRequest::removeIpFromUrlString(_error, request.url());
     //  Negate the connection error as the status
-    if (_status == 0)
-      _status = -1 * (int)reply->error();
+    _status = -1 * (int)reply->error();
     return false;
   }
-
   //  return successfully
   return true;
 }
 
-void HootNetworkRequest::_blockOnReply(QNetworkReply* reply, int timeout)
-{
-  if (reply != NULL)
-  {
-    QTimer timer;
-    //  Qt code to force the use of QNetworkReply to be synchronous
-    QEventLoop loop;
-    QObject::connect(reply, SIGNAL(finished()), &loop, SLOT(quit()));
-    QObject::connect(&timer, SIGNAL(timeout()), &loop, SLOT(quit()));
-    //  Start the timer (convert seconds to miliseconds)
-    timer.start(timeout * 1000);
-    loop.exec();
-    //  Cleanup the timer
-    if (timer.isActive())
-    {
-      //  Reply is finished and the timer is still active, stop the timer
-      timer.stop();
-    }
-    else
-    {
-      //  Timer timed-out, disconnect the reply signal and abort
-      QObject::disconnect(reply, SIGNAL(finished()), &loop, SLOT(quit()));
-      reply->abort();
-    }
-  }
-}
-
```

Comment:
This `blockOnReply()` function wasn't actually blocking correctly so the block actually happened on `readUntil()` once the HTTP response came back.